feat: 리뷰 댓글 CRUD 기능을 구현한다

kiarakim opened the pull request:
## 📄 Summary
> feat: 리뷰 댓글 CRUD 기능을 구현한다

## 🕰️ Actual Time of Completion
> 6시간

## 🙋🏻 More
>

feb-dain committed:
refactor: 토스트 props 위치를 변경한다 (#419)

drunkenhw committed:
feat: Admin 페이지의 충전소 목록을 구현한다  (#421)

* feat: admin 충전소 조회 기능 구현

[#406]

* feat: admin 충전소 조회 api 구현

[#406]

* feat: admin 충전소 상세 조회 기능 구현

[#406]

* feat: admin 충전소 상세 조회 api 구현

[#406]

* feat: admin 충전소 수정 기능 구현

[#406]

* refactor: 리뷰 반영

[#406]

kyw0716 committed:
feat: 간편 로그인 기능을 구현한다 (#420)

* feat: 로그인 기능 구현(미완)

[#397]

* feat: 로그인 기능 구현

[#397]

* refactor: 로그인 관련 로직 함수 분리

[#397]

* feat: 로그인 모달 추가

[#397]

* design: 로그인 모달 디자인 수정

[#397]

* feat: 로그인 redirect 발생시 msw 중지 시키는 기능 구현

[#397]

* refactor: 로그인 에러 처리

- loginUri response를 받아올 때 발생하는 에러에 대한 처리 추가

[#397]

* chore: 웹팩 설정 수정

[#397]

kiarakim committed:
feat: 리뷰 등록 기능 구현

kyw0716 committed:
feat: 개인화 메뉴 관련 mock api를 추가한다 (#423)

* feat: 필터링 조회 mock api 구현

[#422]

* feat: 유저 정보 조회 mock api 구현

[#422]

* feat: 유저 필터 정보 등록 mock api 구현

[#422]

* feat: 유저 필터 정보 조회 mock api 구현

[#422]

* comment: TODO 추가

[#422]

kiarakim committed:
feat: 전체  리뷰 조회 기능 구현

kiarakim committed:
fix: MockBeanInjection 리베이스

kiarakim committed:
feat: 리뷰 수정 기능 구현

kiarakim committed:
test: Controller 테스트 작성

kiarakim committed:
test: restDocs 적용

kiarakim committed:
feat: 리뷰 삭제 기능 구현

kiarakim committed:
refactor: api 수정

kiarakim committed:
refactor: 사용하지 않는 메서드 삭제

kiarakim committed:
refactor: 댓글 수정 및 삭제 API 변경

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/controller/review/ReviewController.java:
```diff
@@ -0,0 +1,51 @@
+            @PathVariable String stationId,
+            @RequestBody CreateReviewRequest createReviewRequest) {
+        reviewService.saveReview(createReviewRequest, stationId, memberId);
+        return ResponseEntity.noContent().build();
```
왜 noContent를 반환하나여? 궁금

kiarakim replied:
보내줄게 없어서요. ok가 좋으려나요

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/controller/review/ReviewController.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @PostMapping("/api/reviews/{reviewId}")
+    public ResponseEntity<Void> updateReview(
```
수정인데 post인 이유가 있나여

drunkenhw replied:
그리고 저희 앞에 api prefix 지워도 될 것 같아요

kiarakim replied:
patch가 좋겠죠 아무래도

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/controller/review/ReviewController.java:
```diff
@@ -0,0 +1,51 @@
+        return ResponseEntity.noContent().build();
+    }
+
+    @DeleteMapping("/api/reviews/{reviewId}")
```
ditto

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/controller/review/dto/ReviewResponse.java:
```diff
@@ -0,0 +1,16 @@
+import java.time.LocalDateTime;
+
+@Builder
+public record ReviewResponse(Long reviewId,
```
얘도 개행 부탁드려요
그리고 빌더 어노테이션을 사용하신 이유가 있나요? 사용하는 곳이 없어보여서요
```suggestion
public record ReviewResponse(
          Long reviewId,
```

kiarakim replied:
중간에 바꾼걸 적용 못햇네요 빼겟습니다

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@Entity
+public class Review extends BaseEntity {
```
equals and hashcode 재정의 부탁드려요

kiarakim replied:
네

kiarakim replied:
근데 왜 다른 도메인엔 없죠?

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+        this.content = content;
+    }
+
+    public void delete() {
+        this.isDeleted = true;
+    }
```
https://velog.io/@jsb100800/Spring-boot-JPA-Soft-Delete-%EA%B5%AC%ED%98%84%ED%95%98%EA%B8%B0
한번 참고해보시는 것이 어떨까요

kiarakim replied:
SqlDelete 좋네요

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+    private String stationId;
+
+    @Column(name = "member_id")
+    private Long memberId;
```
왜 Member 객체, station 객체가 아니라 id만 가지고 있나요?

kiarakim replied:
몰라서요. 적용해볼게여

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
```
얘는 왜 primitive type인가요?

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/exception/review/ReviewExceptionType.java:
```diff
@@ -0,0 +1,40 @@
+    INVALID_CONTENT_NOTNULL(Status.INVALID,2003, "리뷰 내용은 null이 될 수 없습니다"),
+    INVALID_CONTENT_MIN_LENGTH(Status.INVALID,2004, "리뷰 내용은 최소 10자 입니다"),
+    INVALID_CONTENT_MAX_LENGTH(Status.INVALID,2005, "리뷰 내용은 최대 200자 입니다"),
+    UNAUTHORIZED_MEMBER(Status.UNAUTHORIZED, 2006, "리뷰 작성자와 수정자가 다릅니다");
```
cmd + shift+ L
부탁드려요

kiarakim replied:
했어요

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/service/review/ReviewService.java:
```diff
@@ -0,0 +1,108 @@
+        if (content.length() > MAX_CONTENT) {
+            throw new ReviewException(ReviewExceptionType.INVALID_CONTENT_MAX_LENGTH);
+        }
+    }
```
이 검증은 도메인 객체에서 하는 것이 어떤가요

kiarakim replied:
좋은 생각인 것ㄱ 같아요 . 적용했습니다

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/service/review/ReviewService.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    public List<Review> findAllReviews(String stationId, int page) {
+        List<Review> allByStationId = reviewRepository.findAllByStationId(stationId);
```
paging 하지않고 findAll로 하신 이유가 있나요?

kiarakim replied:
페이징을 했다고 생각했느데 다 갖고와서 부분만 리턴하는게 페이징이 아니었네요. 수정했습니다

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/controller/review/ReviewController.java:
```diff
@@ -0,0 +1,51 @@
+        return ResponseEntity.ok(responses);
+    }
+
+    @PostMapping("/api/reviews/{reviewId}")
```
yml에서 api 경로를 붙여줘서 여기에 api 빼도 될 것 같습니다!

kiarakim replied:
아이고. 네 감사합니다

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/controller/review/ReviewController.java:
```diff
@@ -0,0 +1,51 @@
+        return ResponseEntity.noContent().build();
+    }
+
+    @DeleteMapping("/api/reviews/{reviewId}")
```
여기도 빼주세요~

kiarakim replied:
넵

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+import com.carffeine.carffeine.common.domain.BaseEntity;
+import lombok.*;
+
+import javax.persistence.*;
```
import 와일드카드 빼주세요~

kiarakim replied:
아니 이거 왜 설정 적용 안된는거이무ㅜㅜ

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
```
Long이 아니고 long으로 작성하신 이유가 있을까요?

kiarakim replied:
잘못했습니다...

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+    public void delete() {
+        this.isDeleted = true;
+    }
+}
```
마지막에 개행이 없나봐요!

kiarakim replied:
추가햇습니다. 예리하셔라

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/ReviewRepository.java:
```diff
@@ -0,0 +1,14 @@
+
+    Review save(Review review);
+
+    Review findById(Long reviewId);
```
id 기준으로 찾아올 때 Review가 없다면 예외를 어떻게 찾을까요?
`Optional<Review> findById(Long reviewId);`
로 바꿔주면 더 좋을 것 같아요~

kiarakim replied:
네에 감사합니다

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/exception/review/ReviewExceptionType.java:
```diff
@@ -0,0 +1,40 @@
+
+public enum ReviewExceptionType implements ExceptionType {
+
+    INVALID_RATINGS_NOTNULL(Status.INVALID, 2001, "별점은 null이 될 수 없습니다"),
```
`NOT_NULL`이 컨벤션에는 더 맞을 것 같습니다!

kiarakim replied:
네에

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/exception/review/ReviewExceptionType.java:
```diff
@@ -0,0 +1,40 @@
+    INVALID_CONTENT_MIN_LENGTH(Status.INVALID,2004, "리뷰 내용은 최소 10자 입니다"),
+    INVALID_CONTENT_MAX_LENGTH(Status.INVALID,2005, "리뷰 내용은 최대 200자 입니다"),
+    UNAUTHORIZED_MEMBER(Status.UNAUTHORIZED, 2006, "리뷰 작성자와 수정자가 다릅니다");
+
```
불필요한 개행 제거해주세요

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/service/review/ReviewService.java:
```diff
@@ -0,0 +1,108 @@
+        return reviewRepository.save(review);
+    }
+
+    private void validateRequest(CreateReviewRequest request) {
```
이 검증은 서비스말고 Controller에서 들어오는 Dto에 대해서 @Valid로 해보는 건 어떨까요?
애초에 데이터가 검증 후에 서비스에 들어오게요! 혹은 도메인에서도 같이해도 되겠네요

예전에 고민했던 내용인데 저는 주로 @Valid로 Dto에서도 검증하고, 데이터베이스에 들어가는 도메인에서도 한 번 더 해주는데 고민해보시고 자신의 기준을 가져보는 것도 좋을 것 같습니다!

그렇게 된다면 서비스 레이어에서는 불필요한 상수를 관리할 필요도 없고 로직에 집중할 수 있을 것 같아요

kiarakim replied:
네 도메인에서 하도록 했습니다

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/service/review/ReviewService.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    private void validateMember(Review review, Long memberId) {
+        if (!review.getMemberId().equals(memberId)) {
```
`review.getMemberId().equals(memberId)` 이 부분을 도메인 로직으로 뺄 수 있지 않을까요?

kiarakim replied:
네 좋은 것 같아요

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/station/domain/review/ReviewRepositoryTest.java:
```diff
@@ -0,0 +1,63 @@
+        // then
+        assertThat(foundReviews).hasSize(1);
+    }
+}
```
여기도 개행이 없네요!

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/station/domain/review/ReviewRepositoryTest.java:
```diff
@@ -0,0 +1,63 @@
+        reviewRepository.save(review);
+
+        // when
+        String stationId = "ME101010";
```
`stationId`를 given으로 올리고 builder부분에서도 사용하고 when부분에서도 사용할 수 있을 것 같아요

kiarakim replied:
넵 감사합니다

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/service/review/ReviewService.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    public Review updateReview(CreateReviewRequest request, Long reviewId, Long memberId) {
+        validateRequest(request);
+
+        Review review = reviewRepository.findById(reviewId);
+
+        validateMember(review, memberId);
+
+        review.updateReview(request.ratings(), request.content());
+
+        return review;
```
개행 하신 이유가 있나요 조금 보기 불편해서ㅓ요

kiarakim replied:
검증과 검증이 아닌 것들을 나누려고 하다 보니 그렇게 됐습니다. 로직이 바껴서 불편함도 사라지셨을겁니다

drunkenhw commented on backend/src/main/java/com/carffeine/carffeine/station/service/review/dto/CreateReviewRequest.java:
```diff
@@ -0,0 +1,4 @@
+package com.carffeine.carffeine.station.service.review.dto;
+
+public record CreateReviewRequest(Integer ratings, String content) {
```
```suggestion
public record CreateReviewRequest(
Integer ratings,
 String content
 ) {
```

kiarakim replied:
두개여도 줄바꿈을 하는군요

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/station/service/review/ReviewServiceTest.java:
```diff
@@ -0,0 +1,208 @@
+        // then
+        assertThatThrownBy(() -> reviewService.saveReview(request, stationId, memberId))
+                .isInstanceOf(ReviewException.class)
+                .hasMessage("리뷰 내용은 null이 될 수 없습니다");
```
하드코딩된 값으로 검증하는 것이 아닌 ExceptionType.getMessage 를 호출해서 코드로 검증해보는 건 어떨까요?
그렇게 된다면 Exception의 값이 바뀌더라도 테스트는 성공할 것 같습니다

kiarakim replied:
그렇네요. 감사합니다

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/station/service/review/ReviewServiceTest.java:
```diff
@@ -0,0 +1,208 @@
+        // given
+        String stationId = "ME101010";
+        Long memberId = 1L;
+        CreateReviewRequest request = new CreateReviewRequest(4, "이상은 인간에 실로 만천하의 아니다. 대중을 피가 이 보는 심장은 꽃이 역사를 일월과 만천하의 봄바람이다. 피고, 기쁘며, 뼈 길지 속에서 이상이 싸인 황금시대다. 가진 같으며, 풀밭에 이것이다. 보내는 우는 산야에 인생을 부패를 뿐이다. 뜨거운지라, 타오르고 있는 봄바람이다. 뭇 위하여, 역사를 영원히 품었기 것이다. 모래뿐일 것은 부패를 뭇 있으랴? 것은 얼마나 속에 끓는 열락의 것이다. 청춘에서만 생생하며, 영락과 그림자는 힘차게 봄바람이다. 이는 되는 속에서 소리다.이것은 그들의 꽃 끓는다.");
```
ㅋㅋㅋㅋㅋ

drunkenhw replied:
이게 글자가 100이 넘는지 안넘는지는 확실하지 않아보이는데 확실하게 코드로 표현한다던지 그럴 방법이 없을까요?

sosow0212 commented on backend/src/test/java/com/carffeine/carffeine/station/service/review/ReviewServiceTest.java:
```diff
@@ -0,0 +1,208 @@
+        // then
+        assertThat(deletedReview.isDeleted()).isTrue();
+    }
+}
```
여기도 개행 추가해주세요~

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+    private String stationId;
+
+    @Column(name = "member_id")
+    private Long memberId;
```
Member와 연관관계를 안 맺어주신 이유가 있을까요?

kiarakim replied:
잘 몰라서 안했는데 적용했습니다!

drunkenhw commented on backend/src/test/java/com/carffeine/carffeine/station/controller/review/ReviewControllerTest.java:
```diff
@@ -0,0 +1,71 @@
+        // when
+        String stationId = "ME101010";
+        Review review = 선릉역_충전소_리뷰_별4_15글자.get();
+        review.setUpdatedAt(LocalDateTime.of(2023, 8, 11, 23, 34, 8));
```
set Update라는 메서드를 왜 만드시고 사용하신건가요?

kiarakim replied:
The documented type of the field 'reviews[].latestUpdateDate' is String but the actual type is Null
위와 같은 에러가 났는데 review의 최근 수정일에 값이 들어가있지 않아서 직접 넣어줬습니다

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/domain/review/Review.java:
```diff
@@ -0,0 +1,41 @@
+    private long id;
+
+    @Column(name = "station_id")
+    private String stationId;
```
이 부분도 연관관계를 안 맺어주신 이유가 궁금합니다!

kiarakim replied:
그거 잘 몰라서요... JPA 5장 참고해서 다시 해볼겡ㅇㅇ여

drunkenhw commented on backend/src/test/java/com/carffeine/carffeine/station/fixture/review/ReviewFixture.java:
```diff
@@ -0,0 +1,30 @@
+import java.util.List;
+import java.util.function.Supplier;
+
+@SuppressWarnings("NonAsciiCharacters")
+
+public class ReviewFixture {
```
개행 확인해주세요 그리고 전반적으로 코드 포맷팅이 안된거 같아용

drunkenhw commented on backend/src/test/java/com/carffeine/carffeine/station/service/review/ReviewServiceTest.java:
```diff
@@ -0,0 +1,208 @@
+
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@SuppressWarnings("NonAsciiCharacters")
+class ReviewServiceTest extends IntegrationTest {
```
왜 통합테스트로 하신거죠?

drunkenhw replied:
저희 서비스 테스트는 다른 이유가 없다면 단위 테스트를 하기로 정한 것 같아서요

kiarakim replied:
지우고 fake레포지토리 사용했습니다.

drunkenhw commented on backend/src/test/java/com/carffeine/carffeine/station/service/review/ReviewServiceTest.java:
```diff
@@ -0,0 +1,208 @@
+        // then
+        assertThatThrownBy(() -> reviewService.saveReview(request, stationId, memberId))
+                .isInstanceOf(ReviewException.class)
+                .hasMessage("리뷰 내용은 최대 200자 입니다");
```
최대 200자라는 메세지와 테스트 케이스의 이름은 안맞는것가타요

sosow0212 commented on backend/src/main/java/com/carffeine/carffeine/station/service/review/ReviewService.java:
```diff
@@ -0,0 +1,108 @@
+    }
+
+    public List<Review> findAllReviews(String stationId, int page) {
+        List<Review> allByStationId = reviewRepository.findAllByStationId(stationId);
+        return reviewsByPage(allByStationId, page);
```
전체 조회를 하고 페이징 처리를 해주는 것으로 보입니다
다르게 바꿔보는 것은 어떨까요? (ex. page == 0, 5개만 db에서 가져오고 nextPage가 있는지 알려주기)

지금 구조면
1페이지 조회시 `DB에서 모두 조회` -> `코드로 페이징 처리`
2페이지 조회시 `DB에서 모두 조회` -> `코드로 페이징 처리`

이렇게 불필요한 DB 전체 조회가 있네요!

kiarakim replied:
```java
pageable = PageRequest.of(pageable.getPageNumber(), PAGE_ELEMENT_SIZE, Sort.by(Sort.Direction.DESC, "createdAt"));
        Page<Review> reviews = findPageReviews(stationId, pageable);
```

이런 식으로 바꾸봤는데 PAGE_ELEMENT_SIZE는 한 페이지에 나올 리뷰 개수, 즉 10개를 의미합니다.
그래서 조회할 때 쿼리파라미터로 받은 page에 맞춰서 최대 PAGE_ELEMENT_SIZE개를 가져옵니다

sosow0212 reviewed (changes requested):
작업하시느라 수고하셨습니다!
코멘트 확인해주세요~

drunkenhw reviewed (changes requested):
리뷰 남겼습니다~ 수고하셨습니다 👍

drunkenhw commented on backend/src/docs/asciidoc/review.adoc:
```diff
@@ -0,0 +1,19 @@
+=== Response
+
+include::{snippets}/review-controller-test/find-reviews/http-response.adoc[]
+include::{snippets}/review-controller-test/find-reviews/response-fields.adoc[]
```
restdocs 문서는 이것 뿐이가요?

kiarakim replied:
CRUD 다 있어야겠죠?네..

drunkenhw committed:
feat: Admin 페이지에서 보여질 충전소 신고 및 제보 목록을 구현한다 (#427)

* feat: 제보 기능에 실제 member로 변경한다

[#407]

* feat: 충전소 제보 페이징 기능 추가

[#407]

* refactor: 충전소 신고 기능에 회원 추가

[#407]

* test: 충전소 제보 기능 테스트 추가

[#407]

* feat: 충전소 제보 상세 조회 기능 구현

[#407]

* feat: 충전소 제보 확인 기능 추가

[#407]

* feat: 충전소 신고 목록 조회 기능 추가

[#407]

* fix: 잘못된 테스트 수정

[#407]

* fix: 잘못된 테스트 수정

[#407]

* docs: rest docs 문서 업데이트

[#407]

* refactor: 리뷰 반영

[#407]

* refactor: 예외 메세지 수정

[#407]

kiarakim committed:
refactor: Pageable로 페이징 처리

kyw0716 committed:
refactor: 구현된 필터링 컴포넌트에 변경된 api를 적용한다 (#426)

* refactor: 쿼리 키 상수 분리

[#424]

* feat: 서버 충전소 필터링 정보 불러오는 훅 구현

[#424]

* feat: 로그인할 시 유저가 이미 선택했던 필터 정보가 반영되는 기능 구현

[#424]

* refactor: QUERY_KEY postfix → prefix

[#424]

* refactor: StationList의 로딩 스켈레톤이 너무 자주 보이는 문제 수정

- isFetching으로 트리거 → isLoading으로 트리거

[#424]

* feat: 로그인시 유저가 선택했던 필터링 정보가 바로 적용되는 기능 구현

[#424]

* refactor: API 명세 변경에 따른 mock api 수정

[#424]

* comment: 불필요한 주석 삭제

[#424]

* feat: 로그인 후 처리할 로직을 담당하는 컴포넌트 추가

[#424]

* feat: 필터 초기화 기능 추가

[#424]

kiarakim committed:
refactor: 댓글 값 검증 로직 변경

kiarakim committed:
feat: 리뷰 조회 응답에 다음 페이지 전달 기능 구현

gabrielyoon7 committed:
feat: 충전소 리뷰 조회 기능을 구현한다 (#432)

* feat: 메뉴를 좀 더 명확하게 분리

[#412]

* feat: UserRatings 컴포넌트 추가

[#412]

* refactor: 시간 랜덤 생성기 구현

* feat: Review, Reply 타입 추가

* refactor: 데이터 랜덤 생성 함수 일부를 이전

* refactor: 랜덤 생성 시간이 최근 2일 이내로 생성되게 개선

* refactor: 시간 계산 시 초 단위는 `방금 전`으로 표현 개선

* refactor: 랜덤 리뷰가 매번 새로 생성되는 기능 구현

* feat: 충전소 리뷰 랜덤 댓글 생성기 구현

[#412]

* refactor: 충전소 리뷰 헤더 구조 개선

[#412]

* test: UserRatings 스토리북 작성

[#412]

* feat: 리뷰 조회를 위한 카드 구현

[#412]

* feat: 리뷰가 삭제된 경우 보여줄 컴포넌트 구현

[#412]

* refactor: 답글 디자인 개선

[#412]

* feat: 리뷰 및 답글 수정 / 삭제 버튼 구현

[#412]

* feat: 리뷰 평점을 수신하는 기능 구현

[#412]

* feat: 리뷰를 조회하는 기능 구현

[#412]

* refactor: 충전소 카드 디자인 개선

[#412]

* refactor: 삭제된 리뷰는 상세정보의 프리뷰에서 보여지지 않도록 개선

[#412]

* refactor: 수정/삭제 버튼을 확률적으로 나오게 개선 (임시)

[#412]

* refactor: 지도 스크롤 방지

[#412]

* refactor: 잘못된 파일 합병 복구

[#412]

* refactor: 답글 구분선 추가

[#412]

* refactor: 답글 디자인 일부 개선

[#412]

* refactor: 답글 디자인 일부 개선

[#412]

* refactor: 파일명 변경 및 리뷰 모아보는 모달 추가

[#412]

* refactor: 리뷰를 페이지 단위로 반환할 수 있도록 개선

- 무조건 3페이지에서 데이터가 부족해지고 4페이지에서 완전히 끊기도록 구현

[#412]

* feat: 더보기 버튼을 통해 리뷰 조회를 추가로 요청하는 기능 구현

[#412]

* refactor: 무한 스크롤을 담당하는 훅 분리

[#412]

* feat: ReviewCardSkeleton 구현

[#412]

* feat: 삭제된 리뷰에 대한 디자인 개선 및 삭제 표시를 조금 더 명확하게 처리

[#412]

* refactor: 불필요한 태그 제거 등

[#412]

* feat: ReviewPreview 에서도 스켈레톤이 등장하도록 기능 구현

[#412]

* feat: ReviewPreview 에러 대응 기능 구현

[#412]

* refactor: 리뷰 반영

[#412]

kiarakim committed:
fix: 객체 값 비교 로직 변경

kiarakim committed:
refactor: 조회 정렬 변경 및 수정 메서드 변경

kiarakim committed:
docs: 리뷰 등록 RestDocs 작성

kiarakim committed:
docs: 충전소 RestDocs 수정

kiarakim committed:
refactor: Station 연관 관계 매핑

kiarakim committed:
refactor: Member 연관 관계 매핑

kiarakim reviewed (commented):
감사합니다. 근데 이거 왜 pending? 아직 반영 다 안해서 아직은 안해주셔도 돼여
